### PR TITLE
build/tools diff reduce

### PR DESCRIPTION
**Pull request by bsdjhb**

### CODE REVIEW

File: bin/csh/Makefile [hunk 1]
```diff
@@ -121,8 +120,8 @@ build-tools: gethost
 DEPENDOBJS+= gethost
 gethost: gethost.c sh.err.h tc.const.h sh.h ${BUILD_TOOLS_META}
 	@rm -f ${.TARGET}
-	${BUILD_TOOLS_CC} ${BUILD_TOOLS_CFLAGS} ${BUILD_TOOLS_LDFLAGS} \
-	    ${TCSHDIR}/gethost.c -o ${.TARGET}
+	${CC:N${CCACHE_BIN}} -o gethost ${LDFLAGS} ${CFLAGS:C/-DHAVE_ICONV//} \
+	    ${TCSHDIR}/gethost.c
 .if make(build-tools)
```
Comment (bsdjhb) [inline bin/csh/Makefile:125]: This hack here to `rm tc.const.h` is a local diff as well.  I've left it as it probably warrants being removed or upstreamed on its own (I think it is probably orthogonal to the BUILD_TOOLS changes, but maybe the FIXME below is relevant and so this should be removed as part of the BUILD_TOOLS_* commit?)

File: lib/ncurses/tinfo/Makefile [hunk 2]
```diff
@@ -273,16 +273,13 @@ keys.list: MKkeys_list.sh Caps Caps-ncurses
 # Build tools
 DEPENDOBJS+= make_hash make_keys
 build-tools: make_hash make_keys
-BUILD_TOOLS_CFLAGS+=	-I. -I${.CURDIR}/../ncurses
-BUILD_TOOLS_CFLAGS+=	-I${NCURSES_DIR}/ncurses -I${NCURSES_DIR}/include
 
 make_keys: make_keys.c names.c ncurses_def.h ${HEADERS} ${SRCHDRS} ${BUILD_TOOLS_META}
```
Comment (bsdjhb) [inline lib/ncurses/tinfo/Makefile:277]: The addition of SRCHDRS to the dependencies of make_keys and make_hash is also a local diff, but I think it is orthogonal to BUILD_TOOLS_*.  I've no idea if it needs upstreaming?

**Comment by jrtc27** [comment]
BUILD_TOOLS_* date all the way back to
```
commit ba0ca14f52918b36941908a7f25a9e60fac15714
Author: Alex Richardson <Alexander.Richardson@cl.cam.ac.uk>
Date:   Tue May 1 14:07:19 2018 +0100

    [UPSTREAM?] Use new variable BUILD_TOOLS_CC for build-tools
    
    This means I can now do `cheribuild.py cheribsd --cheribsd/subdir bin/sh --clean`
    without getting annoying errors due to trying to execute a MIPS mktokens
    on my x86 host.
    
    I am not sure this is the best approach but at least now running a plain
    make inside a `make buildenv` folder works for more subdirectories.
```
for what it's worth

**Comment by jrtc27** [comment]
It's unclear why that was ever introduced instead of just using CC. Maybe at the time cheribuild was bogusly setting both CC and XCC to CHERI LLVM.

**Comment by bsdjhb** [comment]
Hmm, I guess I should try that test command on my current tree to see if it works.

File: lib/ncurses/tinfo/Makefile (hunk 2 above)
Comment (jrtc27) [reply lib/ncurses/tinfo/Makefile:277]:
```commit b7a1f7fd70367162ae71d75f17bfe1b12e0fac69
Author: Alex Richardson <Alexander.Richardson@cl.cam.ac.uk>
Date:   Fri Dec 14 21:48:21 2018 +0000

    Minor crossbuilding fixes``` is the source of that

**Comment by bsdjhb** [comment]
Humm, it does indeed fail:

```
===> tests/set-e (clean)
+ make -C bin/sh all -j4
sh: ./mksyntax: Exec format error
--- syntax.c ---
*** [syntax.c] Error code 126

make[2]: stopped in /usr/home/john/work/cheri/git/cheribsd/bin/sh
sh: ./mknodes: Exec format error
--- nodes.c ---
*** [nodes.c] Error code 126
```

**Comment by jrtc27** [comment]
Is that ever expected to work though? I'd imagine build-tools needs special treatment even when cross-building on FreeBSD.

**Comment by bsdjhb** [comment]
Doing the revert to see if that alone fixes builds and if so I'll drop it from this PR.

**Pull request merged**